test/rules/no-shadowed-variable/test.ts.lint

```diff
@@ -215,5 +216,105 @@ function creatorFunction(constructor: MyConstructor, filter: MyFunction): void {
 
 for (let i = 0; i < 3; i++) {
     let i = 34;
-        ~       [Shadowed variable: 'i']
+        ~       [err % ('i')]
+}
+
+(function f1() {
+    let f1;
+        ~~ [err % ('f1')]
+})();
+function f2() {
+    const f2 = 0;
+          ~~ [err % ('f2')]
+}
+(function f3() {
+    var f3;
+        ~~ [err % ('f3')]
+})();
+(function f4(f4) {
+             ~~ [err % ('f4')]
+})();
+let f5;
+(function f5(){})()
+          ~~ [err % ('f5')]
+
+class C<C> {
+        ~ [err % ('C')]
+    C<C>(C: C): C {
+      ~ [err % ('C')]
+         ~ [err % ('C')]
+        type C = C;
+             ~ [err % ('C')]
+        namespace C {
+                  ~ [err % ('C')]
+            interface C<C> {}
```

Comment:
@adidahiya I changed the error message to always show `Shadowed name: x`.

Showing a different error message for variables is not really possible:
* We're not tracking if an identifier is a variable
* Variables could merge with classes, interfaces, namespaces ... are they still considered a variable
* And even if we could distinguish the kind of identifiers, it's not clear when to use which message: if a variable shadows a type? if a type shadows a variable? only if a variable shadows another variable?